packages/react-instantsearch-hooks-dom/src/widgets/__tests__/InfiniteHits.test.tsx

```diff
@@ -0,0 +1,214 @@
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                1 - a
+              </strong>
+            </li>
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                2 - b
+              </strong>
+            </li>
+            <li
+              class="ais-InfiniteHits-item"
+            >
+              <strong>
+                3 - c
+              </strong>
+            </li>
+          </ol>
+          <button
+            class="ais-InfiniteHits-loadMore"
+          >
+            Show more results
+          </button>
+        </div>
+      `);
+    });
+  });
+
+  test('show more button refines', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits
+          hitComponent={({ hit: { __position } }) => (
+            <strong>{__position}</strong>
+          )}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(container.querySelectorAll('strong')).toHaveLength(3);
+    });
+
+    act(() => {
+      fireEvent.click(container.querySelector('.ais-InfiniteHits-loadMore')!);
+    });
+
+    await waitFor(() => {
+      expect(container.querySelectorAll('strong')).toHaveLength(6);
+    });
+  });
+
+  test('show previous button refines', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper
+        searchClient={searchClient}
+        initialUiState={{ indexName: { page: 4 } }}
+      >
+        <InfiniteHits
+          hitComponent={({ hit: { __position } }) => (
+            <strong>{__position}</strong>
+          )}
+        />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await wait(0);
+
+    expect(container.querySelectorAll('strong')).toHaveLength(3);
+
+    act(() => {
+      fireEvent.click(
+        container.querySelector('.ais-InfiniteHits-loadPrevious')!
+      );
+    });
+
+    await wait(0);
+
+    expect(container.querySelectorAll('strong')).toHaveLength(6);
+  });
+
+  test('show previous button hidden is showPrevious: false', async () => {
+    const { container } = render(
+      <InstantSearchHooksTestWrapper searchClient={searchClient}>
+        <InfiniteHits showPrevious={false} />
+      </InstantSearchHooksTestWrapper>
+    );
+
+    await waitFor(() => {
+      expect(container.querySelector('.ais-InfiniteHits-loadPrevious')).toBe(
+        null
+      );
+      expect(container).toMatchInlineSnapshot(`
+        <div>
+          <div
+            class="ais-InfiniteHits"
+          >
+            <ol
+              class="ais-InfiniteHits-list"
+            />
+            <button
+              class="ais-InfiniteHits-loadMore ais-InfiniteHits-loadMore--disabled"
+              disabled=""
+            >
+              Show more results
+            </button>
+          </div>
+        </div>
+      `);
+    });
+  });
+});
```

Comment:
Could we also test the `disabled` state of the "Show previous" and "Show more" buttons depending on the loaded page?